src/Tasks/FileState.cs

```diff
@@ -115,9 +115,8 @@ public FileDirInfo(string filename)
                             //
                             // Also, when not under debugger (!) it will give error == 3 for path too long. Make that consistently throw instead.
                             if ((error == 2 /* ERROR_FILE_NOT_FOUND */|| error == 3 /* ERROR_PATH_NOT_FOUND */)
-                                && _filename.Length <= NativeMethodsShared.MAX_PATH)
+                                && _filename.Length <= (int)NativeMethodsShared.OSMaxPathLimit)
```

Comment:
Why the cast?

Reply:
Enum value, maybe we should just split up OSMaxPathLimit into a bool and int value if the only real case to watch for is Windows + 260.